B2B phase 2 changes (#8546)
auth0/docs: articles/architecture-scenarios/implementation/b2b/b2b-logout.md

<%= include('../../_includes/_logout/_introduction.md', { platform: 'b2b' }) %>

## Single Logout

> If you are doing [Federated Logout](#federated-logout) you will likely also want to do Single Logout (SLO).

Reviewer: ```suggestion
If you are doing [Federated Logout](#federated-logout) you will likely also want to do Single Logout (SLO), and there are two main approaches you can take.
```

::: warning
SLO can add complexity to your system, so you need to ensure that you really need it before adding the extra development and maintenance time to your system.
:::

> 

Reviewer: ```suggestion
```

> There are two main approaches you can take to Single Logout:

Reviewer: ```suggestion
```

### Short-lived tokens

> This is by far the simplest approach to Single Logout.  Each application enforces a short time that a user can use the system (5-10 minutes) and then on each action by that user if the time has expired, it will either redirect to Auth0 (for regular web apps), or use [Silent Authentication](https://auth0.com/docs/api-auth/tutorials/silent-authentication) for client side tokens.  This will ensure that eventually (within a minute or two) all other applications will fail to get their refreshed token because the SSO session has been removed through logout.

Reviewer: ```suggestion
This is by far the simplest approach to Single Logout. Each application enforces a short time within which a user can use the system, say, 5-10 minutes. On each action a user performs, if the time has expired then either a redirect to Auth0 (for regular web apps), or [Silent Authentication](https://auth0.com/docs/api-auth/tutorials/silent-authentication) for client side Single Page Applications will be used to obtain new tokens. Ordinarily new tokens will be issued silently due to the Single Sign On (SSO) session. However, after logout, all applications will fail to get new tokens silently because the SSO session will have been removed, and the user will need to re-enter their credentials.
```

> ::: best practice

Reviewer 2: ```suggestion
::: panel Best Practice
```

You want to avoid making too many calls to your Auth0 tenant to avoid rate limiting and poor performance.  A best practice is to only request new tokens if tokens have expired and a user takes an action.  This will avoid applications that are simply open, but not in use, from continually polling for new tokens.
:::

> ::: warning

Reviewer: Having adjacent panels looks messy. Perhaps try to re-arrange this somehow? Maybe break the above paragraph on `Short-lived tokens` into two and add the warning between them?

> ### Build a logout service

Reviewer 3: Do we have a backlog item to build some more guidance around this concept?  Would be nice to provide some additional architectural guidance here.  I have found most customers have a little confusion around this concept, and why this isn't a feature of Auth0 already.

### Build a logout service

> Another technique you can do is build a logout service that can track and destroy application sessions.  Each application will notify the logout service when it creates and removes a session.  This logout service will either have direct access to all application's server side sessions and destroy them directly, or it will have the ability to make a back-channel call to each application to tell the application that it must remove its session.  This technique can be effective because it has low-latency between when a user calls logout and when they are logged out of all applications, but it can add complexity and development time for its creation.  It will also require some way to ensure that new applications added to the system are added to this service.

Reviewer: ```suggestion
Another technique you can use is to build a logout service that can track and destroy application sessions. Each application would notify the logout service when it creates and removes a session. The (logout) service would either have direct access to all application's server side sessions and destroy them directly, or it will have the ability to make a back-channel call to each application to tell the application that it must remove its session. 

This technique can be quite effective as there is low-latency between when a user calls logout, and when they are then logged out of all applications. However it can add complexity and also additional development time for implementation. It will also require some way to ensure that new applications added to the system are added to this service.
```

## Federated Logout

> [Federated User Logout](/logout/guides/logout-idps) may be something that you need to consider for your application.  If you or your customers will be using a third-party IdP (i.e., something other than a [Database Identity Provider](/connections/database)) then the question of whether you need to log the user out of the IdP when they log out of your application is something you will need to answer. The answer depends on what your users would expect. If the application and/or IdP you use is tied closely to a customer organization and a central part of day-to-day operations, then it may be frustrating for users to get logged out of their IdP when they log out of your application. If not, then being logged out of the IdP may be expected, or in some cases even desired. In most B2B scenarios, our customers find that it is preferable *not* to perform federated logout for a user.

Reviewer 3: > The answer depends on what your users would expect
Same comment as before... replace users with **customers**
